File: jscomp/test/a_recursive_type.js
'use strict';

Curry = require("../../lib/js/curry.js");

function g(x) do
  return Curry._1(x[0], x);
end

loop = g(--[ A ]--[g]);

x = --[ A ]--[g];

non_terminate = g(x);

exports.loop = loop;
exports.non_terminate = non_terminate;
--[ loop Not a pure module ]--

File: jscomp/test/a_recursive_type.lua
__console = {log = print};

Curry = require "......lib.js.curry";

function g(x) do
  return Curry._1(x[1], x);
end end

loop = g(--[[ A ]]{g});

x = --[[ A ]]{g};

non_terminate = g(x);

exports = {};
exports.loop = loop;
exports.non_terminate = non_terminate;
return exports;
--[[ loop Not a pure module ]]
